### PR TITLE
Optimize build/run of libc tests

### PR DESCRIPTION
**Pull request by mikbras**
This PR optimizes the **build time** and **run time** of the libc tests, by adding all tests to a single enclave image. This change introduces the **gentests.sh** script to generate wrappers for each test.

The performance change is shown in the table below.

| Measurement | Before           | After           |
| --------------- |---------------|--------------|
| build time       | 35 seconds    | 12 seconds |
| run time          | 129 seconds  | 4 seconds   |
| total time        | 164 seconds  | 16 seconds |

### CODE REVIEW

**Comment by anitagov** [comment]
@mikbras @johnkord @andschwa - We definitely need this PR in the master. PR#1214 was done with the assumption that this PR was in the master.

**Comment by mikbras** [comment]
bors try

**Comment by mikbras** [comment]
try bors

**Comment by mikbras** [comment]
bors try

**Comment by mikbras** [comment]
bors try

File: tests/libc/enc/CMakeLists.txt [hunk 1]
```diff
@@ -1,61 +1,46 @@
...
-
 include(oeedl_file)
-include(${CMAKE_CURRENT_LIST_DIR}/../../../cmake/get_testcase_name.cmake)
+include(add_enclave_executable)
+
+# Include the libc tests wrappers:
+include(../tests.cmake)
+
+# Generate wrappers for libc tests.
+execute_process(
```
Comment (andschwa) [inline tests/libc/enc/CMakeLists.txt]: We might hit a hiccup here where this either caches the generated libc tests, or doesn't cache at all and rebuilds libc tests each time.

But, let's not block the PR on this. I'll investigate once it's in 😄

**Comment by mikbras** [comment]
bors try

File: tests/libc/README.md [hunk 2]
```diff
@@ -1,24 +1,13 @@
...
 ==========
 
-This directory runs each MUSL libc unit test from within an enclave. It does
-this by repeatedly building and runing the enclave located under the 'enc' 
-directory for each unit test found in tests.supported.
+This directory runs the MUSL libc unit tests. It reads **tests.cmake** and 
+generates a wrapper (in the build directory) for each of the tests in this file.
 
-The unit tests are partitioned into three files:
+All unit tests are all listed in **tests.cmake** in the following sections:
```
[the remark this answers is not in the document]
Comment (anitagov) [reply tests/libc/README.md]: @mikbras - Can you please fix this issue?

File: tests/libc/CMakeLists.txt [hunk 3]
```diff
@@ -1,32 +1,10 @@
...
-        elseif (BUILD_TYPE_UPPER MATCHES REL)
-	    if (${testcase} IN_LIST clang_release_skipped_libc_tests)
-		continue ()
-	    endif ()
-        endif()
-    endif()
-    get_testcase_name(${testcase} name "../../3rdparty/musl/libc-test/src/")
-    add_enclave_test(tests/libc/${name} ./host libc_host ./enc libctest-${name}_enc)
-endforeach(testcase)
+add_enclave_test(tests/libc ./host libc_host ./enc libc_enc)
```
[the remark this answers is not in the document]
Comment (anitagov) [reply tests/libc/CMakeLists.txt]: Since fixed by Mike as he merged master to his branch.

**Pull request merged**